src/pricecypher/datasets.py

```diff
@@ -220,6 +232,33 @@ def get_transactions(
         # Map transactions to dicts based on the provided column keys and convert to pandas dataframe.
         return DataFrame.from_records([t.to_dict(scope_keys) for t in transactions])
 
+    def _add_scope_values_for_additional_filters(self, filters, filter_scopes, dataset_id):
```

Comment:
This function mutates the `filters` argument. Instead, I'd suggest removing the `filters` argument altogether and returning the `filters_additional` list. Then, the implementation of the `get_transactions` function can concatenate the return value of this function to the `filters` defined there.

Reply:
done